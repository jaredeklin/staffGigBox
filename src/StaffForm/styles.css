.staff-form {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: auto;
}

.input-name {
  width: 50%;
  border: none;
  border-bottom: 1px solid navy;
  margin: 1rem 0;
}

.add-staff-btn {
  width: 15rem;
  margin: 1rem 0;
  height: 2.5rem;
}

.form-header {
  font-weight: 700;
}
