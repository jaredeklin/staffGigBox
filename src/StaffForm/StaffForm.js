import React, { Component } from 'react';
import './styles.css';
import PropTypes from 'prop-types';

export class StaffForm extends Component {
  constructor(props) {
    super(props);
    this.url = process.env.REACT_APP_API_HOST || 'http://localhost:3000';

    this.state = {
      google_id: this.props.user.uid,
      name: '',
      bar_manager: false,
      ass_bar_manager: false,
      bartender: false,
      barback: false,
      beer_bucket: false
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;

    this.setState({ [name]: value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();

    const response = await fetch(`${this.url}api/v1/staff`, {
      method: 'POST',
      body: JSON.stringify(this.state),
      headers: { 'Content-Type': 'application/json'}
    });

    if ( response.status === 201 ) {
      this.props.addStaff();
    }
  }


  render() {

    return (
      <form className='staff-form' onSubmit={this.handleSubmit}>
        <h4>Add new staff member</h4>
        <input
          className='input-name'
          placeholder='Name'
          name='name'
          value={ this.state.name }
          onChange={ this.handleChange }
        />
        <label className='form-label form-header'>Select all that apply:</label>
        <label className='form-label'>
            Are you a Bar Manager?
          <input type='radio'
            id= 'yes'
            value={ true }
            name='bar_manager'
            onChange={ this.handleChange }

          />
          <label htmlFor='yes'>Yes</label>
          <input
            type='radio'
            id= 'no'
            value={ false }
            name='bar_manager'
            onChange={ this.handleChange }
          />
          <label htmlFor='no'>No</label>
        </label>
        <label className='form-label'>
            Are you an Assistant Bar Manager?
          <input type='radio'
            id= 'yes'
            value={ true }
            name='ass_bar_manager'
            onChange={ this.handleChange }

          />
          <label htmlFor='yes'>Yes</label>
          <input
            type='radio'
            id= 'no'
            value={ false }
            name='ass_bar_manager'
            onChange={ this.handleChange }
          />
          <label htmlFor='no'>No</label>
        </label>
        <label className='form-label'>
            Are you Bartender?
          <input type='radio'
            id= 'yes'
            value={ true }
            name='bartender'
            onChange={ this.handleChange }

          />
          <label htmlFor='yes'>Yes</label>
          <input
            type='radio'
            id= 'no'
            value={ false }
            name='bartender'
            onChange={ this.handleChange }
          />
          <label htmlFor='no'>No</label>
        </label>
        <label className='form-label'>
            Are you Barback?
          <input type='radio'
            id= 'yes'
            value={ true }
            name='barback'
            onChange={ this.handleChange }

          />
          <label htmlFor='yes'>Yes</label>
          <input
            type='radio'
            id= 'no'
            value={ false }
            name='barback'
            onChange={ this.handleChange }
          />
          <label htmlFor='no'>No</label>
        </label>
        <label className='form-label'>
            Do you work the Beer Bucket?
          <input type='radio'
            id= 'yes'
            value={ true }
            name='beer_bucket'
            onChange={ this.handleChange }

          />
          <label htmlFor='yes'>Yes</label>
          <input
            type='radio'
            id= 'no'
            value={ false }
            name='beer_bucket'
            onChange={ this.handleChange }
          />
          <label htmlFor='no'>No</label>
        </label>
        <button className='add-staff-btn'>Add Staff Member</button>
      </form>
    );
  }
}

StaffForm.propTypes = {
  user: PropTypes.object,
  addStaff: PropTypes.func
};
