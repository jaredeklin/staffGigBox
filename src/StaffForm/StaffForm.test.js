import React from 'react';
import { shallow } from 'enzyme';
import { StaffForm } from './StaffForm';

describe('StaffForm', () => {
  let wrapper;
  const mockAddStaff = jest.fn();
  const mockUser = {
    uid: 123456
  };

  beforeEach(() => {
    wrapper = shallow(<StaffForm addStaff={mockAddStaff} user={mockUser} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on handleChange', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Esteban'
      }
    };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('name')).toEqual('Esteban');
  });

  it('should post staff on handle Submit', async () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };

    const mockState = {
      google_id: 123456,
      name: '',
      bar_manager: false,
      ass_bar_manager: false,
      bartender: false,
      barback: false,
      beer_bucket: false
    };

    const expected = [
      'http://localhost:3000/api/v1/staff',
      {
        method: 'POST',
        body: JSON.stringify(mockState),
        headers: {
          'Content-Type': 'application/json'
        }
      }
    ];

    window.fetch = jest.fn(() =>
      Promise.resolve({
        status: 201,
        json: () => Promise.resolve({})
      })
    );
    await wrapper.instance().handleSubmit(mockEvent);
    expect(window.fetch).toHaveBeenCalledWith(...expected);
    expect(mockAddStaff).toHaveBeenCalled();
  });
});
