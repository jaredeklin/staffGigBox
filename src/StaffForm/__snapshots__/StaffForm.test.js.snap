// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaffForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StaffForm
    addStaff={[Function]}
    id="123456"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Add new staff member
</h4>,
        <input
          className="input-name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          value=""
/>,
        <label
          className="form-label form-header"
>
          Select all that apply:
</label>,
        <label
          className="form-label"
>
          Are you a Bar Manager?
          <input
                    id="yes"
                    name="bar_manager"
                    onChange={[Function]}
                    type="radio"
                    value={true}
          />
          <label
                    htmlFor="yes"
          >
                    Yes
          </label>
          <input
                    id="no"
                    name="bar_manager"
                    onChange={[Function]}
                    type="radio"
                    value={false}
          />
          <label
                    htmlFor="no"
          >
                    No
          </label>
</label>,
        <label
          className="form-label"
>
          Are you an Assistant Bar Manager?
          <input
                    id="yes"
                    name="ass_bar_manager"
                    onChange={[Function]}
                    type="radio"
                    value={true}
          />
          <label
                    htmlFor="yes"
          >
                    Yes
          </label>
          <input
                    id="no"
                    name="ass_bar_manager"
                    onChange={[Function]}
                    type="radio"
                    value={false}
          />
          <label
                    htmlFor="no"
          >
                    No
          </label>
</label>,
        <label
          className="form-label"
>
          Are you Bartender?
          <input
                    id="yes"
                    name="bartender"
                    onChange={[Function]}
                    type="radio"
                    value={true}
          />
          <label
                    htmlFor="yes"
          >
                    Yes
          </label>
          <input
                    id="no"
                    name="bartender"
                    onChange={[Function]}
                    type="radio"
                    value={false}
          />
          <label
                    htmlFor="no"
          >
                    No
          </label>
</label>,
        <label
          className="form-label"
>
          Are you Barback?
          <input
                    id="yes"
                    name="barback"
                    onChange={[Function]}
                    type="radio"
                    value={true}
          />
          <label
                    htmlFor="yes"
          >
                    Yes
          </label>
          <input
                    id="no"
                    name="barback"
                    onChange={[Function]}
                    type="radio"
                    value={false}
          />
          <label
                    htmlFor="no"
          >
                    No
          </label>
</label>,
        <label
          className="form-label"
>
          Do you work the Beer Bucket?
          <input
                    id="yes"
                    name="beer_bucket"
                    onChange={[Function]}
                    type="radio"
                    value={true}
          />
          <label
                    htmlFor="yes"
          >
                    Yes
          </label>
          <input
                    id="no"
                    name="beer_bucket"
                    onChange={[Function]}
                    type="radio"
                    value={false}
          />
          <label
                    htmlFor="no"
          >
                    No
          </label>
</label>,
        <button
          className="add-staff-btn"
>
          Add Staff Member
</button>,
      ],
      "className": "staff-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add new staff member",
        },
        "ref": null,
        "rendered": "Add new staff member",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-name",
          "name": "name",
          "onChange": [Function],
          "placeholder": "Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select all that apply:",
          "className": "form-label form-header",
        },
        "ref": null,
        "rendered": "Select all that apply:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Are you a Bar Manager?",
            <input
              id="yes"
              name="bar_manager"
              onChange={[Function]}
              type="radio"
              value={true}
/>,
            <label
              htmlFor="yes"
>
              Yes
</label>,
            <input
              id="no"
              name="bar_manager"
              onChange={[Function]}
              type="radio"
              value={false}
/>,
            <label
              htmlFor="no"
>
              No
</label>,
          ],
          "className": "form-label",
        },
        "ref": null,
        "rendered": Array [
          "Are you a Bar Manager?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "yes",
              "name": "bar_manager",
              "onChange": [Function],
              "type": "radio",
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "htmlFor": "yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "no",
              "name": "bar_manager",
              "onChange": [Function],
              "type": "radio",
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "htmlFor": "no",
            },
            "ref": null,
            "rendered": "No",
            "type": "label",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Are you an Assistant Bar Manager?",
            <input
              id="yes"
              name="ass_bar_manager"
              onChange={[Function]}
              type="radio"
              value={true}
/>,
            <label
              htmlFor="yes"
>
              Yes
</label>,
            <input
              id="no"
              name="ass_bar_manager"
              onChange={[Function]}
              type="radio"
              value={false}
/>,
            <label
              htmlFor="no"
>
              No
</label>,
          ],
          "className": "form-label",
        },
        "ref": null,
        "rendered": Array [
          "Are you an Assistant Bar Manager?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "yes",
              "name": "ass_bar_manager",
              "onChange": [Function],
              "type": "radio",
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "htmlFor": "yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "no",
              "name": "ass_bar_manager",
              "onChange": [Function],
              "type": "radio",
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "htmlFor": "no",
            },
            "ref": null,
            "rendered": "No",
            "type": "label",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Are you Bartender?",
            <input
              id="yes"
              name="bartender"
              onChange={[Function]}
              type="radio"
              value={true}
/>,
            <label
              htmlFor="yes"
>
              Yes
</label>,
            <input
              id="no"
              name="bartender"
              onChange={[Function]}
              type="radio"
              value={false}
/>,
            <label
              htmlFor="no"
>
              No
</label>,
          ],
          "className": "form-label",
        },
        "ref": null,
        "rendered": Array [
          "Are you Bartender?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "yes",
              "name": "bartender",
              "onChange": [Function],
              "type": "radio",
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "htmlFor": "yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "no",
              "name": "bartender",
              "onChange": [Function],
              "type": "radio",
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "htmlFor": "no",
            },
            "ref": null,
            "rendered": "No",
            "type": "label",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Are you Barback?",
            <input
              id="yes"
              name="barback"
              onChange={[Function]}
              type="radio"
              value={true}
/>,
            <label
              htmlFor="yes"
>
              Yes
</label>,
            <input
              id="no"
              name="barback"
              onChange={[Function]}
              type="radio"
              value={false}
/>,
            <label
              htmlFor="no"
>
              No
</label>,
          ],
          "className": "form-label",
        },
        "ref": null,
        "rendered": Array [
          "Are you Barback?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "yes",
              "name": "barback",
              "onChange": [Function],
              "type": "radio",
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "htmlFor": "yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "no",
              "name": "barback",
              "onChange": [Function],
              "type": "radio",
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "htmlFor": "no",
            },
            "ref": null,
            "rendered": "No",
            "type": "label",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Do you work the Beer Bucket?",
            <input
              id="yes"
              name="beer_bucket"
              onChange={[Function]}
              type="radio"
              value={true}
/>,
            <label
              htmlFor="yes"
>
              Yes
</label>,
            <input
              id="no"
              name="beer_bucket"
              onChange={[Function]}
              type="radio"
              value={false}
/>,
            <label
              htmlFor="no"
>
              No
</label>,
          ],
          "className": "form-label",
        },
        "ref": null,
        "rendered": Array [
          "Do you work the Beer Bucket?",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "yes",
              "name": "beer_bucket",
              "onChange": [Function],
              "type": "radio",
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "htmlFor": "yes",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "no",
              "name": "beer_bucket",
              "onChange": [Function],
              "type": "radio",
              "value": false,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "htmlFor": "no",
            },
            "ref": null,
            "rendered": "No",
            "type": "label",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Staff Member",
          "className": "add-staff-btn",
        },
        "ref": null,
        "rendered": "Add Staff Member",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Add new staff member
</h4>,
          <input
            className="input-name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            value=""
/>,
          <label
            className="form-label form-header"
>
            Select all that apply:
</label>,
          <label
            className="form-label"
>
            Are you a Bar Manager?
            <input
                        id="yes"
                        name="bar_manager"
                        onChange={[Function]}
                        type="radio"
                        value={true}
            />
            <label
                        htmlFor="yes"
            >
                        Yes
            </label>
            <input
                        id="no"
                        name="bar_manager"
                        onChange={[Function]}
                        type="radio"
                        value={false}
            />
            <label
                        htmlFor="no"
            >
                        No
            </label>
</label>,
          <label
            className="form-label"
>
            Are you an Assistant Bar Manager?
            <input
                        id="yes"
                        name="ass_bar_manager"
                        onChange={[Function]}
                        type="radio"
                        value={true}
            />
            <label
                        htmlFor="yes"
            >
                        Yes
            </label>
            <input
                        id="no"
                        name="ass_bar_manager"
                        onChange={[Function]}
                        type="radio"
                        value={false}
            />
            <label
                        htmlFor="no"
            >
                        No
            </label>
</label>,
          <label
            className="form-label"
>
            Are you Bartender?
            <input
                        id="yes"
                        name="bartender"
                        onChange={[Function]}
                        type="radio"
                        value={true}
            />
            <label
                        htmlFor="yes"
            >
                        Yes
            </label>
            <input
                        id="no"
                        name="bartender"
                        onChange={[Function]}
                        type="radio"
                        value={false}
            />
            <label
                        htmlFor="no"
            >
                        No
            </label>
</label>,
          <label
            className="form-label"
>
            Are you Barback?
            <input
                        id="yes"
                        name="barback"
                        onChange={[Function]}
                        type="radio"
                        value={true}
            />
            <label
                        htmlFor="yes"
            >
                        Yes
            </label>
            <input
                        id="no"
                        name="barback"
                        onChange={[Function]}
                        type="radio"
                        value={false}
            />
            <label
                        htmlFor="no"
            >
                        No
            </label>
</label>,
          <label
            className="form-label"
>
            Do you work the Beer Bucket?
            <input
                        id="yes"
                        name="beer_bucket"
                        onChange={[Function]}
                        type="radio"
                        value={true}
            />
            <label
                        htmlFor="yes"
            >
                        Yes
            </label>
            <input
                        id="no"
                        name="beer_bucket"
                        onChange={[Function]}
                        type="radio"
                        value={false}
            />
            <label
                        htmlFor="no"
            >
                        No
            </label>
</label>,
          <button
            className="add-staff-btn"
>
            Add Staff Member
</button>,
        ],
        "className": "staff-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add new staff member",
          },
          "ref": null,
          "rendered": "Add new staff member",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-name",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select all that apply:",
            "className": "form-label form-header",
          },
          "ref": null,
          "rendered": "Select all that apply:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Are you a Bar Manager?",
              <input
                id="yes"
                name="bar_manager"
                onChange={[Function]}
                type="radio"
                value={true}
/>,
              <label
                htmlFor="yes"
>
                Yes
</label>,
              <input
                id="no"
                name="bar_manager"
                onChange={[Function]}
                type="radio"
                value={false}
/>,
              <label
                htmlFor="no"
>
                No
</label>,
            ],
            "className": "form-label",
          },
          "ref": null,
          "rendered": Array [
            "Are you a Bar Manager?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "yes",
                "name": "bar_manager",
                "onChange": [Function],
                "type": "radio",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "no",
                "name": "bar_manager",
                "onChange": [Function],
                "type": "radio",
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "no",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Are you an Assistant Bar Manager?",
              <input
                id="yes"
                name="ass_bar_manager"
                onChange={[Function]}
                type="radio"
                value={true}
/>,
              <label
                htmlFor="yes"
>
                Yes
</label>,
              <input
                id="no"
                name="ass_bar_manager"
                onChange={[Function]}
                type="radio"
                value={false}
/>,
              <label
                htmlFor="no"
>
                No
</label>,
            ],
            "className": "form-label",
          },
          "ref": null,
          "rendered": Array [
            "Are you an Assistant Bar Manager?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "yes",
                "name": "ass_bar_manager",
                "onChange": [Function],
                "type": "radio",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "no",
                "name": "ass_bar_manager",
                "onChange": [Function],
                "type": "radio",
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "no",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Are you Bartender?",
              <input
                id="yes"
                name="bartender"
                onChange={[Function]}
                type="radio"
                value={true}
/>,
              <label
                htmlFor="yes"
>
                Yes
</label>,
              <input
                id="no"
                name="bartender"
                onChange={[Function]}
                type="radio"
                value={false}
/>,
              <label
                htmlFor="no"
>
                No
</label>,
            ],
            "className": "form-label",
          },
          "ref": null,
          "rendered": Array [
            "Are you Bartender?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "yes",
                "name": "bartender",
                "onChange": [Function],
                "type": "radio",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "no",
                "name": "bartender",
                "onChange": [Function],
                "type": "radio",
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "no",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Are you Barback?",
              <input
                id="yes"
                name="barback"
                onChange={[Function]}
                type="radio"
                value={true}
/>,
              <label
                htmlFor="yes"
>
                Yes
</label>,
              <input
                id="no"
                name="barback"
                onChange={[Function]}
                type="radio"
                value={false}
/>,
              <label
                htmlFor="no"
>
                No
</label>,
            ],
            "className": "form-label",
          },
          "ref": null,
          "rendered": Array [
            "Are you Barback?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "yes",
                "name": "barback",
                "onChange": [Function],
                "type": "radio",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "no",
                "name": "barback",
                "onChange": [Function],
                "type": "radio",
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "no",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Do you work the Beer Bucket?",
              <input
                id="yes"
                name="beer_bucket"
                onChange={[Function]}
                type="radio"
                value={true}
/>,
              <label
                htmlFor="yes"
>
                Yes
</label>,
              <input
                id="no"
                name="beer_bucket"
                onChange={[Function]}
                type="radio"
                value={false}
/>,
              <label
                htmlFor="no"
>
                No
</label>,
            ],
            "className": "form-label",
          },
          "ref": null,
          "rendered": Array [
            "Do you work the Beer Bucket?",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "yes",
                "name": "beer_bucket",
                "onChange": [Function],
                "type": "radio",
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "htmlFor": "yes",
              },
              "ref": null,
              "rendered": "Yes",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "no",
                "name": "beer_bucket",
                "onChange": [Function],
                "type": "radio",
                "value": false,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "htmlFor": "no",
              },
              "ref": null,
              "rendered": "No",
              "type": "label",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Staff Member",
            "className": "add-staff-btn",
          },
          "ref": null,
          "rendered": "Add Staff Member",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
